File: observability-study/sleuth/producer/src/main/java/com/bso/producer/infra/feign/Service2Feign.java
package com.bso.producer.infra.feign;

import com.bso.producer.model.MyObject;
import org.springframework.cloud.openfeign.FeignClient;
import org.springframework.web.bind.annotation.PostMapping;
import org.springframework.web.bind.annotation.RequestBody;

@FeignClient(
        name = "service2",
        url = "${external-services.service2.url}/object"
)
public interface Service2Feign {
    @PostMapping
    void sendToService2(@RequestBody MyObject body);
}

File: oauth/service1/src/main/kotlin/com/bso/service1/externalcommunication/service2/Service2Feign.kt
package com.bso.service1.externalcommunication.service2

import com.bso.service1.controller.Response
import com.bso.service1.externalcommunication.service2.auth.Service2AuthConfiguration
import org.springframework.cloud.openfeign.FeignClient
import org.springframework.web.bind.annotation.GetMapping

@FeignClient(
    name = "service2Feign",
    url = "http://localhost:8082/hello",
    configuration = [Service2AuthConfiguration::class]
)
interface Service2Feign {

    @GetMapping("protected")
    fun protected(): Response
}